packages/iov-ethereum/src/utils.spec.ts

```diff
@@ -77,6 +86,68 @@ describe("Ethereum utils", () => {
       expect(() => decodeHexQuantity("0xa0xa0xa0xa0xa")).toThrowError(/invalid hex quantity input/);
       expect(() => decodeHexQuantityNonce("0xd3c21bcecceda0ffffff")).toThrowError(/Input not in int53 range/);
       expect(() => decodeHexQuantityNonce("0xde0b6b3a763ffff")).toThrowError(/Input not in int53 range/);
+      expect(() => decodeHexQuantityNonce("0x0400")).toThrowError(/invalid hex quantity input/);
+    });
+  });
+
+  describe("encodeQuantity", () => {
+    it("verify valid inputs", () => {
+      let encQtyHex;
+      encQtyHex = encodeQuantity(10000000000000);
+      expect(encQtyHex).toEqual("0x9184e72a000");
+      encQtyHex = encodeQuantity(1024);
+      expect(encQtyHex).toEqual("0x400");
+      encQtyHex = encodeQuantity(65);
+      expect(encQtyHex).toEqual("0x41");
+      encQtyHex = encodeQuantity(2);
+      expect(encQtyHex).toEqual("0x2");
+      encQtyHex = encodeQuantity(0);
+      expect(encQtyHex).toEqual("0x0");
+    });
+
+    it("throws error for invalid inputs", () => {
+      expect(() => encodeQuantity(NaN)).toThrowError(/Input is not a number/);
+      expect(() => encodeQuantity(12345678901234567890)).toThrowError(/Input is not a safe integer/);
+    });
+  });
+
+  describe("encodeQuantityString", () => {
+    it("verify valid inputs", () => {
+      let encQtyHex;
+      encQtyHex = encodeQuantityString("10000000000000");
+      expect(encQtyHex).toEqual("0x9184e72a000");
+      encQtyHex = encodeQuantityString("1024");
+      expect(encQtyHex).toEqual("0x400");
+      encQtyHex = encodeQuantityString("65");
+      expect(encQtyHex).toEqual("0x41");
+      encQtyHex = encodeQuantityString("2");
+      expect(encQtyHex).toEqual("0x2");
+      encQtyHex = encodeQuantityString("0");
+      expect(encQtyHex).toEqual("0x0");
+    });
+
+    it("throws error for invalid inputs", () => {
+      expect(() => encodeQuantityString("1234abc")).toThrowError(/Input is not a valid string number/);
+    });
+  });
+
+  describe("stringDataToHex", () => {
+    it("verify valid inputs", () => {
+      let dataToHex;
+      dataToHex = stringDataToHex("A");
+      expect(dataToHex).toEqual("0x41");
+      dataToHex = stringDataToHex("\0B\0");
+      expect(dataToHex).toEqual("0x004200");
+      dataToHex = stringDataToHex("");
+      expect(dataToHex).toEqual("0x");
```

Comment:
Since this is used for memo, please also add characters that are more than one byte, e.g.

> ⇉ (Rightwards Paired Arrows, U+21c9) takes 2 bytes in UTF-8